ui/lib/utils/asyncDebounce.ts

```diff
@@ -0,0 +1,21 @@
+import { debounce, DebounceSettings } from 'lodash'
+
+export function asyncDebounce<T extends (...args: any) => any>(
+  func: T,
+  wait?: number,
+  options?: DebounceSettings
+): T {
+  const debounced = debounce(
+    (resolve, reject, args) => {
+      func(...args)
+        .then(resolve)
```

Comment:
According to your type definition `func` is `T` where `T extends (...args: any) => any`. In this case, I don' think `func(..).then` is ensured to exist.